src/cpp/session/modules/build/SessionBuild.cpp

```diff
@@ -317,6 +317,9 @@ class Build : boost::noncopyable,
       else
          core::system::unsetenv(&environment, "RSTUDIO_CONSOLE_WIDTH");
 
+      // this runs in the build pane
+      core::system::setenv(&environment, "RSTUDIO_PANE", "build");
```

Comment:
So when running in build, the pid of the parent process should be the same as `RSTUDIO_PID`